src/base_alloc/base_alloc.c

```diff
@@ -134,7 +134,10 @@ static void *ba_os_alloc_annotated(size_t pool_size) {
 }
 
 umf_ba_pool_t *umf_ba_create(size_t size) {
-    size_t chunk_size = ALIGN_UP(size, MEMORY_ALIGNMENT);
+    size_t chunk_size = ALIGN_UP_SAFE(size, MEMORY_ALIGNMENT);
+    if (!chunk_size) {
```

Comment:
IMHO in case of numeric values you should use:
```c
 if (chunk_size == 0)
```

(instead of ` if (!chunk_size)`) - also in all places below.

Reply:
Done